### ISSUE TITLE
[OpenAPI] Support using `Default` trait when derive `Object`

### ISSUE THREAD

**Issue by duskmoon314** [poem-web/poem#369]
## Description of the feature

Currently, only the fields of `Object` has the `default` parameter. And it can only be used for some types that already implement `Default`, or have a function to generate the default value.

It might be more convenient if we could define `Default` for structs and use it.

Notice: I don't know if this is possible. Have taken a simple look at the macro of `Object`, it takes fields' attributes one by one and generates codes.

## Code example (if possible)

Suppose rust codes like this:

```rust
#[derive(Object)]
#[oai(default)]
struct MyObj {
  str: String,
  int: isize
}

impl Default for MyObj {
  fn default() -> Self {
    Self {
      str: "some_string".to_string(),
      int: 32,
    }
  }
}
```

**Comment by sunli829** [comment]
Absolutely, I'll add it later.

**Comment by sunli829** [comment]
Released in `v2.0.10`

https://github.com/poem-web/poem/blob/05d568e4654b672aafb61488bcfa1d0662607e6f/poem-openapi/tests/object.rs#L823

**Comment by duskmoon314** [comment]
Amazing! Thanks!
